include/boost/math/special_functions/hypergeometric_1F1.hpp

```diff
@@ -0,0 +1,207 @@
+
+///////////////////////////////////////////////////////////////////////////////
+//  Copyright 2014 Anton Bikineev
+//  Copyright 2014 Christopher Kormanyos
+//  Copyright 2014 John Maddock
+//  Copyright 2014 Paul Bristow
+//  Distributed under the Boost
+//  Software License, Version 1.0. (See accompanying file
+//  LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef _BOOST_HYPERGEOMETRIC_1F1_HPP_
+  #define _BOOST_HYPERGEOMETRIC_1F1_HPP_
+
+#include <boost/math/policies/policy.hpp>
+#include <boost/math/policies/error_handling.hpp>
+#include <boost/math/special_functions/detail/hypergeometric_series.hpp>
+#include <boost/math/special_functions/detail/hypergeometric_asym.hpp>
+#include <boost/math/special_functions/detail/hypergeometric_rational.hpp>
+#include <boost/math/special_functions/detail/hypergeometric_1f1_recurrence.hpp>
+#include <boost/math/special_functions/detail/hypergeometric_pade.hpp>
+#include <boost/math/special_functions/detail/hypergeometric_1f1_bessel.hpp>
+#include <boost/math/special_functions/detail/hypergeometric_1f1_scaled_series.hpp>
+
+namespace boost { namespace math { namespace detail {
+
+   // check when 1f1 series can't decay to polynom
+   template <class T>
+   inline bool check_hypergeometric_1f1_parameters(const T& a, const T& b)
+   {
+      BOOST_MATH_STD_USING
+
+         if ((b <= 0) && (b == floor(b)))
+         {
+            if ((a >= 0) || (a < b) || (a != floor(a)))
+               return false;
+         }
+
+      return true;
+   }
+
+   template <class T, class Policy>
+   inline T hypergeometric_1f1_imp(const T& a, const T& b, const T& z, const Policy& pol)
+   {
+      BOOST_MATH_STD_USING // exp, fabs, sqrt
+
+         static const char* const function = "boost::math::hypergeometric_1f1<%1%,%1%,%1%>(%1%,%1%,%1%)";
+
+      if ((z == 0) || (a == 0))
```

Comment:
This branch is a set of measure zero. Does it actually improve performance?

Reply:
Probably force of habit to pick off trivial special cases first - if you think it's better without, try removing it and seeing what happens ;)

Reply:
I think we need a separable PDE which has 1F1 as a component. Then there will be an associated series of the form 1F1(a_n,b_n,x)g_n(y) (something like that) and it will give us some idea of whether practical use generates parameter values with sufficient frequency to make this worthwhile.

I'll go ahead and write a benchmark to find out what the performance impact is; you're probably right that it's negligible.

Reply:
It's not only performance - the result should be exactly 1 in that situation, regardless of the other arguments.